burnman/mineral.py

```diff
@@ -64,7 +65,17 @@ class which derives from the equation_of_state base class.
             raise Exception("Please derive your method from object (see python old style classes)")
 
         if self.method is not None and type(new_method) is not type(self.method):
-            warnings.warn('Overriding database equation of state. From ' + self.method.__class__.__name__ + ' to ' + new_method.__class__.__name__, stacklevel=2)
+
+            # Warn user that they are changing the EoS
+            warnings.warn('Overriding database equation of state from ' + self.method.__class__.__name__ + ' to ' + new_method.__class__.__name__ + '. This does not overwrite any mineral attributes other than temperature and pressure. Non-implemented attributes of the old method will remain unchanged unless they are explicitly changed by the user.', stacklevel=2)
+
+            # Set new EoS
+            self.params['equation_of_state']=equation_of_state
```

Comment:
let's say I have a mineral with "SLB2" parameters and I now do
```
mat.set_method("mgd3")
...
mat.set_method("mgd2")
...
mat.set_method("slb2")
```

Will currently produce:
```
Warning: change from SLB2 to MGD3.
Warning: change from SLB2 to MGD2.
(no warning)
```

With your change we will get:
```
Warning: change from SLB2 to MGD3.
Warning: change from MGD3 to MGD2.
Warning: change from MGD2 to SLB2.
```

I think the last ```set_method``` is the "natural" method and should not produce a warning. Maybe we should rephrase it to say "Warning, you are changing the method to __ even though the material is defined as to be used with ___"?